src/data-workspace/category-combo-table-body-pivoted/category-combo-table-body-pivoted.js

```diff
@@ -0,0 +1,174 @@
+            dataElements,
+        }
+
+        const rowsMatrix = generateFormMatrix(options, displayOptions)
+
+        return (
+            <>
+                {rowsMatrix.map((row, id /** todo: find suitable id */) => {
+                    return (
+                        <TableRow key={id}>
+                            {row.map((fieldInRow) => {
+                                if (
+                                    fieldInRow.type === 'columnHeader' ||
+                                    fieldInRow.type === 'rowHeader'
+                                ) {
+                                    return (
+                                        <TableCellHead
+                                            key={fieldInRow.id}
+                                            className={classNames(
+                                                [
+                                                    styles.pivotedHeader,
+                                                    styles.noWrap,
+                                                    styles[
+                                                        fieldInRow.metadataType
+                                                    ],
+                                                ],
+                                                {
+                                                    [styles.dataElementRowHeader]:
+                                                        fieldInRow.metadataType ===
+                                                        'dataElement',
+                                                }
+                                            )}
+                                            colSpan={fieldInRow.colSpan}
+                                            rowSpan={fieldInRow.rowSpan}
+                                        >
+                                            {fieldInRow.displayFormName !==
+                                                'default' &&
+                                                fieldInRow.displayFormName}
+                                        </TableCellHead>
+                                    )
+                                }
+
+                                if (fieldInRow.type === 'empty') {
+                                    return (
+                                        <TableCell
+                                            className={[
+                                                styles.categoryNameHeader,
+                                                styles.noWrap,
+                                            ]}
+                                            key={fieldInRow.id}
+                                            colSpan={fieldInRow.colSpan}
+                                            rowSpan={fieldInRow.rowSpan}
+                                        />
+                                    )
+                                }
+
+                                if (fieldInRow.type === 'de') {
```

Comment:
one thing about my implementation of the matrix generation is that it also generates the data element entry fields. The previous implementation had this as a component concern, and I think that was cleaner - so as part of the consolidating these components, I will also follow the previous approach.

Reply:
Sounds good. Yeah I think the data-structure could just return the Ids, and the component would get the dataElement in component-level? But it doesn't matter too much I think.

But I think it makes sense that it generates the structure for them - since now everything is intertwined. It's not as simple as looping over dataElements and categoryOptionsCombos anymore. One row is not always equal to the cocs.length anylonger, so I'm not sure how easy it is to do this on component level?
I guess the structure would have the rows (dataElements and categories), and the headers, and could loop from that info. But I think this is fine for now.